addons/lunch/static/src/js/lunch.js

```diff
@@ -0,0 +1,41 @@
+openerp.lunch  = function(instance) {
+    var QWeb = instance.web.qweb;
+    var _t = instance.web._t;
+
+    instance.lunch.PreviousOrder = instance.web.form.AbstractField.extend({
+        set_value: function (value_) {
+            var self = this;
+            var values = JSON.parse(value_);
+            this.pref_ids = values['pref_ids'];
+            if (this.pref_ids != 0) {
+                self.categories = values['categories'];
+                self.currency = values['currency'];
+            }
+         },
+        update_lines: function (pref_id, records) {
+            var self = this;
+            _.each(self.categories, function(category) {
+                _.each(category, function(prod_obj, prod_id) {
+                    _.each(prod_obj, function(prod_detail, prod_note) {
+                        if (prod_detail.pref_id == parseInt(pref_id)) {
+                            if(prod_note == 'False') { prod_note = ''; }
+                            records.push([0, 0, {'product_id': parseInt(prod_id), 'note': prod_note, 'price': prod_detail.price, 'supplier': prod_detail.supplier, 'state': prod_detail.state}]);
+                            self.field_manager.set_values({order_line_ids: records});
+                        }
+                    });
+                });
+            });
+        },
+        render_value: function() {
+            var self = this;
+            this.$el.html(QWeb.render("lunch.PreviousOrder", {widget: self}));
+            this.$el.find('button.oe_button_plus, button.oe_button_add').click( function(event) {
+                var pref_id = $(event.currentTarget).attr('id');
+                var records = self.field_manager.get_field_value('order_line_ids');
+                self.update_lines(pref_id, records);
+            });
+         },
+    });
+
+    instance.web.form.widgets.add('previous_order', 'instance.lunch.PreviousOrder');
```

Comment:
@jem-odoo 
As per suggestion of tde we avoid rpc calls for fetching data,
and about using the functional field, we used non-stored char type field bcz here we only need of  jsonify data
If we add m2m field then note that m2m will require separate rpc call to fetch data from ids